feat(ast): derive `GetAddress` on all enum types

overlookmotel opened the pull request:
Implement `GetAddress` on all AST enums where all variants are boxed. Part of #7339.

Boshen commented:
### Merge activity

* **Nov 25, 7:12 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

overlookmotel committed:
feat(ast): derive `GetAddress` on all enum types (#7472)

Implement `GetAddress` on all AST enums where all variants are boxed. Part of #7339.
